onnxruntime_extensions/tools/add_HuggingFace_CLIPImageProcessor_to_model.py

```diff
@@ -0,0 +1,122 @@
+
+    if 'rescale' in extend_config:
+        # default 255
+        steps.append(ImageBytesToFloat())
+
+    if 'tochw' in extend_config and extend_config['tochw']:
+        steps.append(ChannelsLastToChannelsFirst())
+
+    if 'normalize' in extend_config:
+        norm_arg = extend_config['normalize']
+        mean_std = [(0.485, 0.229), (0.456, 0.224), (0.406, 0.225)]
+        layout = 'CHW'
+        if isinstance(norm_arg, dict):
+            mean_std = norm_arg.get('mean_std', [(0.485, 0.229), (0.456, 0.224), (0.406, 0.225)])
+            layout = norm_arg.get('layout', 'CHW')
+
+        steps.append(Normalize(mean_std, layout=layout))
+
+    steps.append(Unsqueeze([0]))  # add batch dim
+
+    return steps
+
+
+def clip_image_processor(model_file: Path, output_file: Path, onnx_opset: int = 16, extend_config: dict = None):
+    """
+    Used for models like stable-diffusion. should be compatible with 
+    https://github.com/huggingface/transformers/blob/main/src/transformers/models/clip/image_processing_clip.py
+
+    It applies to 'CLIP' image processor, and aligns with the HuggingFace class name.
+
+    A typical usage example is used in Stable diffusion.
+
+    :param model_file: The input model file path.
+    :param output_file: The output file path, where the finalized model saved to.
+    :param onnx_opset: The opset version of onnx model, default(16).
+    :param extend_config: The extended config for the model.
+    """
+    # Normalize config key to lower case
+    if extend_config is not None:
+        extend_config = {k.lower(): v for k, v in extend_config.items()}
+    else:
+        extend_config = {"resize": None, "center_crop": None, "rescale": None, "tochw": True, "normalize": True }
+    # Load model
+    model = onnx.load(str(model_file.resolve(strict=True)))
+    inputs = [create_named_value("image", onnx.TensorProto.UINT8, ["num_bytes"])]
+
+    pipeline = PrePostProcessor(inputs, onnx_opset)
+
+    preprocessing = image_processor(extend_config=extend_config)
+
+    pipeline.add_pre_processing(preprocessing)
+
+    new_model = pipeline.run(model)
+    onnx.save_model(new_model, str(output_file.resolve()))
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        os.path.basename(__file__),
+        description="""Add CLIPImageProcessor to a model.
+
+        The updated model will be written in the same location as the original model, 
+        with '.onnx' updated to '.with_clip_processor.onnx'
+
+        Example usage:
+            object detection:
+            - python -m onnxruntime_extensions.tools.add_HuggingFace_CLIPImageProcessor_to_model model.onnx  
+        """,
+    )
+
+    parser.add_argument(
+        "--opset", type=int, required=False, default=16,
+        help="ONNX opset to use. Minimum allowed is 16. Opset 18 is required for Resize with anti-aliasing.",
+    )
+    
+    parser.add_argument(
+        "--extend_config", type=str, required=False, default=None,
```

Comment:
Instead of making the user create a json file can we simplify and use equivalent arguments to https://github.com/huggingface/transformers/blob/35eac0df75c692c5b93c12f7eaf3279cab8bd7ce/src/transformers/models/clip/image_processing_clip.py#L56? 

I would expect the usage is typically a one-off for the user, so the overhead of creating a config file may not provide much benefit at the cost of them having to create a correctly formatted json file.

Reply:
Sure.
Aligned the args with huggingFace ones.